plugins/generic/metadataExport/MetadataExportHandler.inc.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+/**
+ * @file plugins/generic/metadataExport/MetadataExportHandler.inc.php
+ *
+ * Copyright (c) 2013-2015 Simon Fraser University Library
+ * Copyright (c) 2003-2015 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @package plugins.generic.metadataExport
+ * @class MetadataExportHandler
+ */
+
+import('classes.handler.Handler');
+
+class MetadataExportHandler extends Handler {
+
+	/* @var journal Journal object - the current journal */
+	var $journal;
+	
+	/* @var articles Array - the articles of the exported issue/journal */
+	var $articles;
+	
+	/* @var fileName String - the name of the export file */
+	var $fileName;
+	
+	/* @var publishedArticleDao PublishedArticleDAO object */
+	var $publishedArticleDao;
+	
+	/* @var issueDao IssueDAO object */
+	var $issueDao;
+	
+	/* @var metadataExportPlugins Array - all plugins of the metadataexport category */
+	var $metadataExportPlugins;
+	
+	/* @var exportFunction String - the function defining the scope of the export (article/issue/journal) */
+	var $exportFunction;
+	
+	/* @var exportPluginName String - the name of the current export plugin */
+	var $exportPluginName;
+	
+	/* @var exportPlugin MetadataExportPlugin object - the current export plugin object */
+	var $exportPlugin;
+	
+	/* @var elementId Int - the id of the current article/issue */
+	var $elementId;
+	
+	
+	/**
+	 * @function export -  Call the respective export function (for article, issue or journal).
+	 * @param $args Array
+	 */
+	function export($args) {
+		$scope = Request::getUserVar('exportScope');
+		
+		$this->exportPluginName = Request::getUserVar('exportPluginName');
+		$this->elementId = Request::getUserVar('elementId');
+		$this->metadataExportPlugins = PluginRegistry::loadCategory('generic/metadataExport/metadataExportFormats');
+		$this->exportPlugin = $this->metadataExportPlugins[$this->exportPluginName];
+		$this->journal = Request::getJournal();
+		$this->publishedArticleDao = DAORegistry::getDAO('PublishedArticleDAO');
+		$this->issueDao = DAORegistry::getDAO('IssueDAO');
+
+		switch ($scope) {
+			case 'exportArticle':
+				$this->_exportArticle();
+				break;
+			case 'exportIssue':
+				$this->_exportIssue();
+				break;
+			case 'exportJournal':
+				$this->_exportJournal();
+				break;
+			default:
+				assert(false);
+		}
+	}
+	
+	/**
+	 * @function _exportArticle - Export current article
+	 */
+	function _exportArticle() {
+		$articleId = $this->elementId;
+		$article = $this->publishedArticleDao->getPublishedArticleByArticleId($articleId);
```

Comment:
(Generally there don't seem to be any access controls here. What about unpublished content? Might need testing.)